src/components/OSCALLoader.test.js

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import OSCALLoader from "./OSCALLoader";
+
+describe("The OSCAL Loader", () => {
+  let loader;
+  let oscalObjectType;
+
+  const loaderUrl = "http://loader";
+  const sspUrl = "http://ssp";
+  const newSspUrl = "http://ssp1";
+
+  const changeURL = jest.fn();
+  const handleUrlChange = jest.spyOn(React, "useState");
+
+  test("OSCALurl overwrites default url", () => {
+    // GIVEN
+    oscalObjectType = { defaultUrl: loaderUrl };
+
+    loader = (
+      <OSCALLoader
+        renderForm={<div />}
+        hasDefaultUrl
+        oscalObjectType={oscalObjectType}
+        oscalUrl={sspUrl}
+      />
+    );
+
+    // WHEN
+    render(loader);
+    expect(loader.props.oscalUrl).toEqual(sspUrl);
+
+    handleUrlChange.mockImplementation((u) => [u, changeURL]);
+    loader.props.oscalUrl = newSspUrl;
```

Comment:
I would like to be able to somehow mock the oscalUrl state change. From what I have seen, this is done a lot by simulating a click. The loader is not a button so that is made a bit difficult.